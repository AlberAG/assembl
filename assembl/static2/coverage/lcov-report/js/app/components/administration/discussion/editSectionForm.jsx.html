<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/administration/discussion/editSectionForm.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">js/app/components/administration/discussion</a> editSectionForm.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">43.75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.11% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">11.11% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">46.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/15</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { Translate, I18n } from 'react-redux-i18n';
import { FormGroup, Checkbox, OverlayTrigger, Button } from 'react-bootstrap';
import { getEntryValueForLocale } from '../../../utils/i18n';
import FormControlWithLabel from '../../common/formControlWithLabel';
import { deleteSectionTooltip, upTooltip, downTooltip } from '../../common/tooltips';
import {
  updateSectionTitle,
  updateSectionUrl,
  toggleExternalPage,
  deleteSection,
  moveSectionUp,
  moveSectionDown
} from '../../../actions/adminActions/adminSections';
&nbsp;
type EditSectionFormProps = {
  url: string,
  type: string,
  title: string,
  nbSections: number,
  index: number,
  handleTitleChange: Function,
  handleUrlChange: Function,
  toggleExternalPageField: Function,
  handleDeleteClick: Function,
  handleDownClick: Function,
  handleUpClick: Function
};
&nbsp;
const DumbEditSectionForm = ({
  url,
  type,
  title,
  nbSections,
  index,
  handleTitleChange,
  toggleExternalPageField,
  handleUrlChange,
  handleDeleteClick,
  handleDownClick,
  handleUpClick
}: EditSectionFormProps) =&gt; {
  const titlePh = I18n.t('administration.sections.titlePh');
  const urlPh = I18n.t('administration.sections.urlPh');
  const hasUrl = url !== null;
  return (
    &lt;div className="form-container"&gt;
      &lt;div className="title left"&gt;
        {`${index + 1}. `}
        {title || <span class="branch-1 cbranch-no" title="branch not covered" >I18n.t('administration.sections.custom')}</span>
      &lt;/div&gt;
      &lt;div className="right"&gt;
        {type !== 'HOMEPAGE' ? (
<span class="branch-0 cbranch-no" title="branch not covered" >          &lt;div className="inline"&gt;</span>
            {index &lt; nbSections - 1 ? (
              &lt;OverlayTrigger placement="top" overlay={downTooltip}&gt;
                &lt;Button onClick={handleDownClick} className="admin-icons"&gt;
                  &lt;span className="assembl-icon-down-bold grey" /&gt;
                &lt;/Button&gt;
              &lt;/OverlayTrigger&gt;
            ) : null}
            {index &gt; 1 ? (
              &lt;OverlayTrigger placement="top" overlay={upTooltip}&gt;
                &lt;Button onClick={handleUpClick} className="admin-icons"&gt;
                  &lt;span className="assembl-icon-up-bold grey" /&gt;
                &lt;/Button&gt;
              &lt;/OverlayTrigger&gt;
            ) : null}
          &lt;/div&gt;
        ) : null}
        {type === 'CUSTOM' ? (
<span class="branch-0 cbranch-no" title="branch not covered" >          &lt;OverlayTrigger placement="top" overlay={deleteSectionTooltip}&gt;</span>
            &lt;Button onClick={handleDeleteClick} className="admin-icons"&gt;
              &lt;span className="assembl-icon-delete grey" /&gt;
            &lt;/Button&gt;
          &lt;/OverlayTrigger&gt;
        ) : null}
      &lt;/div&gt;
      &lt;div className="clear" /&gt;
      {type !== 'HOMEPAGE' ? (
<span class="branch-0 cbranch-no" title="branch not covered" >        &lt;FormControlWithLabel label={titlePh} onChange={handleTitleChange} type="text" value={title} /&gt;</span>
      ) : null}
      {type === 'HOMEPAGE' ? (
        &lt;FormGroup&gt;
          &lt;Checkbox checked={hasUrl} onChange={toggleExternalPageField}&gt;
            &lt;Translate value="administration.sections.externalPage" style={{ lineHeight: '20px' }} /&gt;
          &lt;/Checkbox&gt;
        &lt;/FormGroup&gt;
      ) : <span class="branch-1 cbranch-no" title="branch not covered" >null}</span>
      {(type === 'HOMEPAGE' &amp;&amp; hasUrl) || <span class="branch-2 cbranch-no" title="branch not covered" >type === 'CUSTOM' </span>? (
        &lt;FormControlWithLabel label={urlPh} onChange={handleUrlChange} type="text" value={url} /&gt;
      ) : <span class="branch-1 cbranch-no" title="branch not covered" >null}</span>
      &lt;div className="separator" /&gt;
    &lt;/div&gt;
  );
};
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate, { id, editLocale }) =&gt; {
  const section = <span class="cstat-no" title="statement not covered" >state.admin.sections.sectionsById.get(id);</span>
<span class="cstat-no" title="statement not covered" >  return {</span>
    locale: state.i18n.locale, // for I18n.t()
    url: section.get('url'),
    type: section.get('type'),
    order: section.get('order'),
    title: getEntryValueForLocale(section.get('titleEntries'), editLocale, '')
  };
};
&nbsp;
const mapDispatchToProps = <span class="fstat-no" title="function not covered" >(d</span>ispatch, { id, editLocale }) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  handleTitleChange: <span class="fstat-no" title="function not covered" >e </span>=&gt; <span class="cstat-no" title="statement not covered" >dispatch(updateSectionTitle(id, editLocale, e.target.value)),</span>
  handleUrlChange: <span class="fstat-no" title="function not covered" >e </span>=&gt; <span class="cstat-no" title="statement not covered" >dispatch(updateSectionUrl(id, e.target.value)),</span>
  toggleExternalPageField: <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >dispatch(toggleExternalPage(id)),</span>
  handleDeleteClick: <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >dispatch(deleteSection(id)),</span>
  handleUpClick: <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >dispatch(moveSectionUp(id)),</span>
  handleDownClick: <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >dispatch(moveSectionDown(id))</span>
});
&nbsp;
export { DumbEditSectionForm };
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(DumbEditSectionForm);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
