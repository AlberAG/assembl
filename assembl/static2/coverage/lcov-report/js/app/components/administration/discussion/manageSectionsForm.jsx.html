<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/administration/discussion/manageSectionsForm.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">js/app/components/administration/discussion</a> manageSectionsForm.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/10</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { OverlayTrigger } from 'react-bootstrap';
import { List } from 'immutable';
&nbsp;
import SectionTitle from '../sectionTitle';
import EditSectionForm from './editSectionForm';
import { addSectionTooltip } from '../../common/tooltips';
import * as actions from '../../../actions/adminActions/adminSections';
import { createRandomId } from '../../../utils/globalFunctions';
&nbsp;
type ManageSectionFormProps = {
  sections: List&lt;string&gt;,
  editLocale: string,
  createSection: Function
};
&nbsp;
const DumbManageSectionsForm = ({ sections, editLocale, createSection }: ManageSectionFormProps) =&gt; (
  &lt;div className="admin-box"&gt;
    &lt;SectionTitle title={I18n.t('administration.sections.sectionsTitle')} annotation={I18n.t('administration.annotation')} /&gt;
    &lt;div className="admin-content"&gt;
      &lt;form&gt;
        {sections.map((id, index) =&gt; (
          &lt;EditSectionForm key={id} id={id} index={index} editLocale={editLocale} nbSections={sections.size} /&gt;
        ))}
        &lt;OverlayTrigger placement="top" overlay={addSectionTooltip}&gt;
          &lt;div onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >createSection(sections.size)}</span> className="plus margin-l"&gt;
            +
          &lt;/div&gt;
        &lt;/OverlayTrigger&gt;
      &lt;/form&gt;
    &lt;/div&gt;
  &lt;/div&gt;
);
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
  const { sectionsInOrder } = <span class="cstat-no" title="statement not covered" >state.admin.sections;</span>
<span class="cstat-no" title="statement not covered" >  return {</span>
    editLocale: state.admin.editLocale,
    locale: state.i18n.locale, // for I18n.t()
    sections: sectionsInOrder
  };
};
&nbsp;
const mapDispatchToProps = <span class="fstat-no" title="function not covered" >di</span>spatch =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  createSection: <span class="fstat-no" title="function not covered" >(n</span>extOrder) =&gt; {
    const newId = <span class="cstat-no" title="statement not covered" >createRandomId();</span>
<span class="cstat-no" title="statement not covered" >    return dispatch(actions.createSection(newId, nextOrder));</span>
  }
});
&nbsp;
export { DumbManageSectionsForm };
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(DumbManageSectionsForm);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
