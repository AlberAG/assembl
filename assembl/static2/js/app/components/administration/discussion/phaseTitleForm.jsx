// @flow
import React from 'react';
import { connect } from 'react-redux';
import { OverlayTrigger, Button, FormGroup } from 'react-bootstrap';
import { I18n, Translate } from 'react-redux-i18n';
import classnames from 'classnames';
import { updatePhaseTitle, deletePhase, movePhaseUp, movePhaseDown } from '../../../actions/adminActions/timeline';
import FormControlWithLabel from '../../common/formControlWithLabel';
import { deletePhaseTooltip, upTooltip, downTooltip } from '../../common/tooltips';
import { getEntryValueForLocale } from '../../../utils/i18n';

type PhaseTitleFormProps = {
  id: string,
  title: string,
  editLocale: string,
  handleTitleChange: Function,
  handleDeleteClick: Function,
  handleUpClick: Function,
  handleDownClick: Function,
  phaseIndex: number,
  numberOfPhases: number,
  hasConflictingDates: boolean
}

export const DumbPhaseTitleForm = ({
  id,
  title,
  editLocale,
  handleTitleChange,
  handleDeleteClick,
  handleUpClick,
  handleDownClick,
  phaseIndex,
  numberOfPhases,
  hasConflictingDates
}: PhaseTitleFormProps) => {
  const phaseLabel = I18n.t('administration.timelineAdmin.phaseLabel');
  const isTitleEmpty = title.length === 0;
  const isFirst = phaseIndex === 1;
  const isLast = phaseIndex === numberOfPhases;
  const phaseSideTitleClassNames = classnames('phase-side-title', { 'phase-side-title-low': isTitleEmpty });
  const deleteButtonClassNames = classnames('admin-icons', { 'delete-button-high': isTitleEmpty });
  return (
    <div className="flex phase-title-form">
      <Translate value="administration.timelineAdmin.phase" count={phaseIndex} className={phaseSideTitleClassNames} />
      <FormGroup validationState={hasConflictingDates ? 'warning' : null}>
        <FormControlWithLabel
          key={`phase-${id}`}
          label={`${phaseLabel} ${editLocale.toUpperCase()}`}
          onChange={handleTitleChange}
          type="text"
          value={title}
          required
          className={hasConflictingDates ? 'warning' : null}
        /></FormGroup>
      <div className="flex">
        {!isLast ? (
          <OverlayTrigger placement="top" overlay={downTooltip}>
            <Button onClick={handleDownClick} className={isFirst ? 'admin-icons end-items' : 'admin-icons'}>
              <span className="assembl-icon-down-small grey" />
            </Button>
          </OverlayTrigger>
        ) : null}
        {!isFirst ? (
          <OverlayTrigger placement="top" overlay={upTooltip}>
            <Button onClick={handleUpClick} className={isLast ? 'admin-icons end-items' : 'admin-icons'}>
              <span className="assembl-icon-up-small grey" />
            </Button>
          </OverlayTrigger>
        ) : null}
        <OverlayTrigger placement="top" overlay={deletePhaseTooltip}>
          <Button onClick={handleDeleteClick} className={deleteButtonClassNames}>
            <span className="assembl-icon-delete grey" />
          </Button>
        </OverlayTrigger>
      </div>
    </div>
  );
};

const mapStateToProps = (state, { id, editLocale, phaseIndex }: PhaseTitleFormProps) => {
  const { phasesById } = state.admin.timeline;
  const phase = phasesById.get(id);
  const start = phase.get('start');
  const end = phase.get('end');
  const nextPhaseId = state.admin.timeline.phasesInOrder.toJS()[phaseIndex];
  const previousPhaseId = state.admin.timeline.phasesInOrder.toJS()[phaseIndex - 2];
  const previousPhaseEnd = phasesById.getIn([previousPhaseId, 'end']);
  const nextPhaseStart = phasesById.getIn([nextPhaseId, 'start']);
  const hasConflictingDates = start && start.isBefore(previousPhaseEnd) || end && end.isAfter(nextPhaseStart);
  return {
    title: getEntryValueForLocale(phase.get('titleEntries'), editLocale, ''),
    hasConflictingDates: hasConflictingDates
  };
};

const mapDispatchToProps = (dispatch, { id, editLocale }: PhaseTitleFormProps) => ({
  handleTitleChange: e => dispatch(updatePhaseTitle(id, editLocale, e.target.value)),
  handleDeleteClick: () => dispatch(deletePhase(id)),
  handleUpClick: () => dispatch(movePhaseUp(id)),
  handleDownClick: () => dispatch(movePhaseDown(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(DumbPhaseTitleForm);