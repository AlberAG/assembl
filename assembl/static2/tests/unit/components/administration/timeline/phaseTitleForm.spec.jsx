import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { DumbPhaseTitleForm } from '../../../../../js/app/components/administration/discussion/phaseTitleForm';

describe('PhaseTitleForm component', () => {
  const handleTitleChangeSpy = jest.fn(() => {});
  const handleDeleteClickSpy = jest.fn(() => {});

  it('should render a form to update the phase\'s title', () => {
    const props = {
      id: '123',
      title: 'Une phase int√©ressante',
      editLocale: 'fr',
      handleTitleChange: handleTitleChangeSpy,
      handleDeleteClick: handleDeleteClickSpy,
      phaseIndex: 1
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<DumbPhaseTitleForm {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});