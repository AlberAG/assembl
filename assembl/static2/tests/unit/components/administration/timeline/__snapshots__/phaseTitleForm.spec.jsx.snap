// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseTitleForm component should render a form to update the phase's title 1`] = `
<div
  className="flex phase-title-form"
>
  <Translate
    className="phase-side-title"
    count={1}
    value="administration.timelineAdmin.phase"
  />
  <FormControlWithLabel
    componentClass={undefined}
    disabled={undefined}
    formControlProps={undefined}
    helperText=""
    helperUrl=""
    id={undefined}
    label="Phase title FR"
    labelAlwaysVisible={false}
    name={undefined}
    onChange={[Function]}
    required={true}
    type="text"
    validationErrors={undefined}
    value="Une phase intÃ©ressante"
  />
  <div
    className="flex"
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="deletePhaseTooltip"
          placement="right"
        >
          <Translate
            value="administration.timelineAdmin.deletePhase"
          />
        </Tooltip>
      }
      placement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="admin-icons"
        disabled={false}
        onClick={[Function]}
      >
        <span
          className="assembl-icon-delete grey"
        />
      </Button>
    </OverlayTrigger>
  </div>
</div>
`;
